rest/user_api_test.go

```diff
@@ -1352,3 +1352,59 @@ func TestUserXattrAvoidRevisionIDGeneration(t *testing.T) {
 
 	assert.NotEqual(t, syncData2.CurrentRev, syncData3.CurrentRev)
 }
+
+func TestGetUserCollectionAccess(t *testing.T) {
+
+	rt := NewRestTester(t, nil)
+	defer rt.Close()
+
+	// Create a user with collection metadata
+	userPayload := `{
+		"email":"alice@couchbase.com", 
+		"password":"letmein", 
+		"admin_channels":["foo", "bar"],
+		"collection_access": {
+			"scope1": {
+				"collection1": {
+					"admin_channels":["foo", "bar1"]
+				}
+			}
+		}
+	}`
+
+	putResponse := rt.SendAdminRequest("PUT", "/db/_user/alice", userPayload)
+	RequireStatus(t, putResponse, 201)
+
+	getResponse := rt.SendAdminRequest("GET", "/db/_user/alice", "")
+	RequireStatus(t, getResponse, 200)
+	log.Printf("response:%s", getResponse.Body.Bytes())
+	var responseConfig auth.PrincipalConfig
+	err := json.Unmarshal(getResponse.Body.Bytes(), &responseConfig)
+	require.NoError(t, err)
+	collectionAccess, ok := responseConfig.CollectionAccess["scope1"]["collection1"]
+	require.True(t, ok)
+	assert.Equal(t, channels.BaseSetOf(t, "foo", "bar1"), collectionAccess.ExplicitChannels_)
+	// TODO: computed channels requires authenticator populated with collection set, pending CBG-2266
+	//assert.Equal(t, channels.BaseSetOf(t), collectionAccess.Channels_)
```

Comment:
Is this now pending RestTester collections? And does this (and the below test cases) make sense to spin off?